### PR TITLE
feat(RELEASE-951): integrate Checkton for linting

### PR DESCRIPTION
**Pull request by seanconroy2021**
This updats the linting workflow by adding Checkton. Checkton will review code for linting errors.
_**Please see other PR [here](https://github.com/hacbs-release/app-interface-deployments/pull/127)**_
Changes:
* Added a step to the workflow to run Checkton for PR.
* Configured Checkton to upload the SARIF report
*

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -43,3 +43,20 @@ jobs:
...
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Run Checkton
+        id: checkton
+        uses: chmeliik/checkton@main
+        with:
+          fail-on-findings: false
+      - name: Upload SARIF File
```
Comment (johnbieren) [inline .github/workflows/lint.yaml:62]: What does uploading the file get us? Any idea?

Comment (seanconroy2021) [reply .github/workflows/lint.yaml:62]: @johnbieren The Checkton only finds the linting problems, so uploading the results to SARIF will create warning messages on the PR.  Better explanation  [ here  PR ](https://github.com/seanconroy2021/shellchecktest/pull/6)

Comment (johnbieren) [reply .github/workflows/lint.yaml:62]: ah, okay cool

**Review by johnbieren — DISMISSED** [review]
/ok-to-test

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -43,3 +43,20 @@ jobs:
...
+      - name: Run Checkton
+        id: checkton
+        uses: chmeliik/checkton@v0.1.2
+        with:
+          fail-on-findings: false
+      - name: Upload SARIF File
+        uses: github/codeql-action/upload-sarif@v3
+        with:
+          sarif_file: ${{ steps.checkton.outputs.sarif }}
+          category: checkton
```
Comment (seanconroy2021) [inline .github/workflows/lint.yaml]: Fixing the hidden Char

**Review by johnbieren — DISMISSED** [review]

File: .github/workflows/lint.yaml (hunk 2 above)
Comment (johnbieren) [reply .github/workflows/lint.yaml]: Looks like it is back?

Comment (seanconroy2021) [reply .github/workflows/lint.yaml]: It fixed now 👍🏻

**Review by johnbieren — APPROVED** [review]

**Pull request merged**